Add documentation for trusted publishers

leobalter opened the pull request:
This adds new documentation for trusted publishers, to be released soon.

## References

Related to https://github.com/npm/cli/pull/8336
Replaces #1673 (now using a branch without a fork)

steiza commented on content/packages-and-modules/securing-your-code/trusted-publishers.mdx:
```diff
@@ -0,0 +1,269 @@
+title: Trusted publishing for npm packages
+---
+
+Trusted publishing allows you to publish npm packages directly from your CI/CD workflows using [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/) authentication, eliminating the need for long-lived npm tokens. This feature implements the [Trusted Publishers standard](https://repos.openssf.org/trusted-publishers-for-all-package-repositories) from the Open Source Security Fou ...
```
Nit: technically it's not a standard (a standard would come from a standards body, like the IETF).

steiza replied:
SGTM!

steiza reviewed (approved):
This is great! One minor nit.

leobalter commented:
I rebased the branch - squashing the previous commits into 3abe6fb0f96938ceb70356c9ba5b56de5ff7cb21 - and added a little change to replace minimun cli version required for oidc from 11.5.0 to 11.5.1 here: 95c4ece0ece8fe200a90e6be863f0f0f5d93b3fb

karenjli commented on content/packages-and-modules/securing-your-code/trusted-publishers.mdx:
```diff
@@ -0,0 +1,269 @@
+- [About npm provenance](./generating-provenance-statements)
+- [OpenSSF Trusted Publishers specification](https://repos.openssf.org/trusted-publishers-for-all-package-repositories)
+- [GitHub Actions OIDC documentation](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect)
+- [GitLab CI/CD OIDC documentation](https://docs.gitlab.com/ee/ci/cloud_services/)
```
```suggestion
- [GitLab CI/CD OIDC documentation](https://docs.gitlab.com/ee/ci/cloud_services/)
- [API documentation for exchanging OIDC ID token for npm registry token](https://api-docs.npmjs.com/#tag/registry.npmjs.org/operation/exchangeOidcToken)

```
This adds the token exchange API documentation link
